### PR TITLE
DEVPROD-5296: Support jumping to shared or failing line with sectioning enabled

### PR DESCRIPTION
**Pull request by SupaJoon**
DEVPROD-5296

### Description
These code changes open the section of a shared or failing line before jumping to it on page load. The section is set to open during section initialization.

### CODE REVIEW

File: apps/parsley/src/hooks/useSections/index.ts [hunk 1]
```diff
@@ -100,12 +110,42 @@ export const useSections = ({
         return currentState;
       });
     },
-    [sectionState],
+    [],
   );
+
+  const openSectionContainingLineNumber = useCallback(
+    ({ lineNumber }: { lineNumber: number | number[] }) => {
```
[the remark this answers is not in the document]
Comment (SupaJoon) [reply apps/parsley/src/hooks/useSections/index.ts:117]: Searching logs will pass an array of line numbers to open.

File: apps/parsley/src/components/LogPane/index.tsx [hunk 2]
```diff
@@ -15,25 +15,49 @@ interface LogPaneProps {
...
+        shareLine ??
+        (settings.jumpToFailingLineEnabled ? failingLine : undefined);
+      if (
+        sectioningEnabled &&
+        jumpToLine !== undefined &&
+        !performedOpen.current
+      ) {
+        openSectionContainingLineNumber({ lineNumber: jumpToLine });
+        performedOpen.current = true;
+      }
```
[the remark this answers is not in the document]
Comment (SupaJoon) [reply apps/parsley/src/components/LogPane/index.tsx]: I adjusted the logic to handle opening the initial open section during section parsing. This actually led to better UX by making the loading flow less "jumpy" with one less render.

File: apps/parsley/src/hooks/useSections/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (SupaJoon) [reply apps/parsley/src/hooks/useSections/index.ts]: I ended up removing this function and will reintroduce in a later PR regarding search.

**Review by minnakt — COMMENTED** [review]
i really love this refactor!

File: apps/parsley/src/hooks/useSections/index.ts [hunk 3]
```diff
@@ -111,14 +117,23 @@ export const useSections = ({
   };
 };
 
-const populateSectionState = (sectionData: SectionData): SectionState => {
+const populateSectionState = (
+  sectionData: SectionData,
+  openSectionContainingLine: number | undefined,
```
Comment (minnakt) [inline apps/parsley/src/hooks/useSections/index.ts:122]: just wondering if this parameter name `openSectionContainingLine` should match the one above `onInitOpenSectionContainingLine`!

**Review by minnakt — APPROVED** [review]
looks GREAT to me! 🤠

File: apps/parsley/src/hooks/useSections/index.ts (hunk 3 above)
Comment (SupaJoon) [reply apps/parsley/src/hooks/useSections/index.ts:122]: I think it should be different. populateSectionState is a function that returns a sectionState with default values based on certain inputs. It doesn't really know whether the lineNumber is an initialization value or onChange value. This function may be reused later.

**Pull request merged**